### ISSUE TITLE
Support code generation via kube_codegen.sh when the repository path is outside the package name

### ISSUE THREAD

**Issue by mimowo** [kubernetes/code-generator#165]
In particular, the script only works when the kueue repository is cloned into the "sigs.k8s.io" folder, so that it matches its package name prefix : sigs.k8s.io/kueue. 
See more details here: https://github.com/kubernetes-sigs/kueue/pull/1151#discussion_r1336826593
It was also reported as a complication for jobset: https://github.com/kubernetes-sigs/jobset/issues/305

**Comment by sportshead** [comment]
Workaround for multi-level packages (e.g. `github.com/sportshead/codegen-demo` instead of just `sigs.k8s.io/kueue`:
```bash
# TODO: remove the workaround when the issue is solved in code-generator
# (https://github.com/kubernetes/code-generator/issues/165).
# kube_codegen.sh expects this layout:
# .
# └── github.com
#     └── sportshead
#         └── codegen-demo
# We can use soft links in order to fake this layout, such that
# ./github.com/sportshead/codegen-demo resolves to ././../codegen-demo, or ./.
ln -s . github.com
ln -s .. sportshead
trap "rm github.com && rm sportshead" EXIT
```

**Comment by thockin** [comment]
All the PATH and GOPATH stuff should be resolved now!

/close

**Comment by caozhuozi** [comment]
> All the PATH and GOPATH stuff should be resolved now!
> 
> /close

Hi, @thockin  from which tag the issue is solved? Could you share any commit related to the fix?

**Comment by thockin** [comment]
I thought it was fixed in v0.30.0-beta.0 but there are other issues with go.mod - it MIGHT work for you, else it should be fixed in beta.1 - let me know if not?
